### PR TITLE
DOC: Gitpod documentation

### PR DESCRIPTION
**Pull request by noatamir**
xref #48107  

Documentation for using the development Gitpod.

### CODE REVIEW

**Comment by noatamir** [comment]
I got some warning about the title capitalization in pre-commit. But it was relating to `Gitpod` and `Liveserve`, and words after the colon, which I wasn't use about the style guide there. Is that a problem? What's your recommendation here? I would prefer the brands to be capitalized in the titles.

**Comment by jorisvandenbossche** [comment]
For the capitalization, there is a list of exceptions that get ignored here:  https://github.com/pandas-dev/pandas/blob/e41b6d7827720df90a51ff05caa689333d7e02af/scripts/validate_rst_title_capitalization.py#L21

File: doc/source/development/contributing_gitpod.rst [hunk 1]
```diff
@@ -0,0 +1,282 @@
...
+you do not have an account yet, you will need to create one first.
+
+Head over to the `Gitpod`_ website and click on the **Continue with GitHub**
+button. You will be redirected to the GitHub authentication page.
+You will then be asked to install the `Gitpod GitHub app <https://github.com/marketplace/gitpod-io>`_.
+
+Make sure to select **All repositories** access option to avoid issues with
+permissions later on. Click on the green **Install** button
+
+.. image:: ./gitpod-imgs/installing-gitpod-io.png
```
Comment (jorisvandenbossche) [inline doc/source/development/contributing_gitpod.rst:37]: This image seems to be missing in the branch? (and some of the others as well, there are only 2 images committed)

File: doc/source/development/contributing_environment.rst [hunk 2]
```diff
@@ -188,6 +191,17 @@ Enable Docker support and use the Services tool window to build and manage image
 run and interact with containers.
 See https://www.jetbrains.com/help/pycharm/docker.html for details.
 
+Option 4: using GitPod
```
Comment (noatamir) [inline doc/source/development/contributing_environment.rst:194]:
```suggestion
Option 4: using Gitpod
```

File: doc/source/development/contributing_gitpod.rst [hunk 3]
```diff
@@ -0,0 +1,277 @@
...
+#. Browse to the `pandas repository on GitHub`_ and `create your own fork`_.
+
+#. Browse to your fork. Your fork will have a URL like
+   https://github.com/noatamir/pandas-dev, except with your GitHub username in place of
+   ``noatamir``.
+
+Starting Gitpod
+---------------
+Once you have authenticated to Gitpod through GitHub, you can install the
+`Gitpod browser extension <https://www.gitpod.io/docs/browser-extension>`_
```
Comment (phofl) [inline doc/source/development/contributing_gitpod.rst:51]: Maybe add a note that safari is not supported?

File: doc/source/development/contributing_gitpod.rst [hunk 4]
```diff
@@ -0,0 +1,277 @@
...
+#. Once the build is complete, you will be directed to your workspace,
+   including the VSCode editor and all the dependencies you need to work on
+   pandas. The first time you start your workspace, you will notice that there
+   might be some actions running. This will ensure that you have a development
+   version of pandas installed.
+
+#. When your workspace is ready, you can :ref:`test the build<contributing.running_tests>` by
+   entering::
+
+      $ python pytest pandas
```
Comment (phofl) [inline doc/source/development/contributing_gitpod.rst:78]: This does not work for me, have to do

```
python -m pytest pandas
```

File: doc/source/development/contributing_gitpod.rst [hunk 5]
```diff
@@ -0,0 +1,277 @@
...
+--------------------
+Gitpod uses VSCode as the editor. If you have not used this editor before, you
+can check the Getting started `VSCode docs`_ to familiarize yourself with it.
+
+Your workspace will look similar to the image below:
+
+.. image:: ./gitpod-imgs/gitpod-workspace.png
+   :alt: Gitpod workspace screenshot
+
+.. note::  By default, VSCode initializes with a light theme. You can change to
```
Comment (phofl) [inline doc/source/development/contributing_gitpod.rst:91]: Somehow mine did not, installed it fresh to try gitpod. Got the dark theme right from the beginning

**Comment by phofl** [comment]
I think it makes sense to add I as an exception

**Review by phofl — APPROVED** [review]

**Comment by phofl** [comment]
Merging, we have a sprint tomorrow and would like to reference to this.

Please ping with suggestions, if we should improve anything

**Comment by phofl** [comment]
thx @noatamir

**Pull request merged**